### PR TITLE
/university endpoint

### PR DESCRIPTION
**Pull request by terencehuynh**
Create university API endpoints

### CODE REVIEW

File: app/controllers/index.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+const requireDirectory = require('require-directory');
+
+module.exports = requireDirectory(module);
```
Comment (hugomd) [inline app/controllers/index.js]: Not sure I agree with this, but at least it's clean 👌

File: app/controllers/University.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
+const Boom = require('boom');
+const models = require('../models');
+const validators = require('../validators');
+
+const NOT_FOUND_ERROR = 'Cannot find university with that id';
+
+// [GET] /university
+exports.index = {
```
Comment (terencehuynh) [inline app/controllers/University.js]: Probably. Happy to change it, what do you suggest @hugomd?

File: app/controllers/University.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    models.University.findById(id)
+      .then((result) => {
+        if (!result) {
+          return res(Boom.notFound(NOT_FOUND_ERROR));
+        }
+        return res(result);
+      });
+  },
+  validate: {
+    params: validators.Generic.uuid,
```
Comment (terencehuynh) [inline app/controllers/University.js]: Noted.

File: app/controllers/University.js [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+  },
+};
+
+// [POST] /university
+exports.post = {
+  handler: (req, res) => {
+    const payload = req.payload;
+    console.log(payload);
```
Comment (terencehuynh) [inline app/controllers/University.js]: Removed.

File: app/controllers/index.js (hunk 1 above)
Comment (terencehuynh) [inline app/controllers/index.js]: Putting it similar to models to make the code look cleaner. Happy to remove and refactor it.

File: app/controllers/University.js [hunk 5]
```diff
@@ -0,0 +1,104 @@
+const Boom = require('boom');
+const models = require('../models');
+const validators = require('../validators');
+
+const NOT_FOUND_ERROR = 'Cannot find university with that id';
+
+// [GET] /university
+exports.index = {
+  handler: (req, res) => {
+    models.University.findAll()
```
Comment (terencehuynh) [inline app/controllers/University.js]: Noted.

File: app/controllers/University.js (hunk 2 above)
Comment (hugomd) [inline app/controllers/University.js]: @terencehuynh I tend to do `exports.getUniversityById` — be _really_ explicit

File: app/controllers/University.js [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+
+// [GET] /university/{id}
+exports.get = {
+  handler: (req, res) => {
+    const id = req.params.id;
+
+    models.University.findById(id)
+      .then((result) => {
+        if (!result) {
+          return res(Boom.notFound(NOT_FOUND_ERROR));
```
Comment (hugomd) [inline app/controllers/University.js]: @terencehuynh That works !

File: app/controllers/University.js [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+  handler: (req, res) => {
+    const id = req.params.id;
+
+    models.University.findById(id)
+      .then((university) => {
+        if (!university) {
+          return res(Boom.notFound(NOT_FOUND_ERROR));
+        }
+
+        return university.destroy({
```
Comment (hugomd) [inline app/controllers/University.js]: Do we want to maybe do a soft-delete?

File: app/controllers/index.js (hunk 1 above)
Comment (hugomd) [inline app/controllers/index.js]: Yeah, that's true. I think it's probably easier this way in the long run though. I'll leave it up to your discretion 👍

File: app/controllers/University.js (hunk 7 above)
Comment (terencehuynh) [inline app/controllers/University.js]: For universities? That's probably a discussion we need to have outside this review. Happy to do a quick chat.

**Pull request merged**